src/main/java/org/spongepowered/common/mixin/core/world/biome/MixinBiomeMesa.java

```diff
@@ -52,9 +52,11 @@
 
     @Shadow @Final private boolean brycePillars;
     @Shadow @Final private boolean hasForest;
+    private boolean vanillaCompat = true;
 
     @Override
     public void buildPopulators(World world, SpongeBiomeGenerationSettings gensettings) {
+        this.vanillaCompat = false;
```

Comment:
This doesn't work with multi-world, as there is only one biome instance for the entire server, so long as any world is using the normal generation all worlds will have the decoration cancelled. A flag set from the generator for whether or not the current generation is calling a modded override would work.